src/Framework/TestCase.php

```diff
@@ -57,6 +57,11 @@
 {
     private const LOCALE_CATEGORIES = [\LC_ALL, \LC_COLLATE, \LC_CTYPE, \LC_MONETARY, \LC_NUMERIC, \LC_TIME];
 
+    private const DEPENDENCY_PREFIX = 'Dependency_';
+
+    /** @var TestResult \PHPUnit\Framework\TestResult */
+    private static $DEPENDENCY_TASK_RESULTS;
```

Comment:
This data structure is only needed for the edge case of automatically running dependencies located in another class. Does this need to be a class-wide variable?
Also: why `static`, it belongs to an actual `TestCase` instance?

Reply:
It's static for optimization reasons. 

Meaning, if multiple functions require this same dependency, the results will be used from this class property instead of running dependency again and again, maybe even 100 times, before it will run the test.